source/GameInterface/Services/Settlements/Patches/LastThreatTimeSettlementPatch.cs

```diff
@@ -0,0 +1,55 @@
+﻿using Common;
+using Common.Messaging;
+using Common.Util;
+using Coop.Mod.Extentions;
+using GameInterface.Extentions;
+using GameInterface.Policies;
+using GameInterface.Services.Settlements.Messages;
+using HarmonyLib;
+using System;
+using System.Collections.Generic;
+using System.Reflection;
+using System.Text;
+using TaleWorlds.CampaignSystem;
+using TaleWorlds.CampaignSystem.Settlements;
+
+namespace GameInterface.Services.Settlements.Patches;
+
+
+[HarmonyPatch(typeof(Settlement))]
+internal class LastThreatTimeSettlementPatch
+{
+    [HarmonyPatch(nameof(Settlement.LastThreatTime), MethodType.Setter)]
+    [HarmonyPrefix]
+    private static bool LastThreatTimePrefix(ref Settlement __instance, ref CampaignTime value)
+    {
+        if (AllowedThread.IsThisThreadAllowed()) return true;
+        if (CallOriginalPolicy.IsOriginalAllowed()) return true;
+
+        if (ModInformation.IsClient) return false;
```

Comment:
Change this to match up to line 42 of https://github.com/Bannerlord-Coop-Team/BannerlordCoop/blob/development/source/GameInterface/Services/Armies/Patches/ArmyLifetimePatches.cs#L42